src/libcore/cell.rs

```diff
@@ -1512,16 +1512,14 @@ impl<T: ?Sized> UnsafeCell<T> {
     }
 }
 
-// TODO: correct stable attribute
-#[stable(feature = "rust1", since = "1.0.0")]
-impl<T: Clone> Clone for UnsafeCell<T> {
+#[stable(feature = "unsafe_cell_copy_clone_conservative", since = "1.31.0")]
+impl<T: Clone + Copy> Clone for UnsafeCell<T> {
     fn clone(&self) -> UnsafeCell<T> {
         UnsafeCell::new(self.value.clone())
```

Comment:
This shouldn't call `clone()`, that is still unknown code executing.

It should make a `Copy` instead.

Reply:
Is there an explict way to call copy? or is this acceptable?

```rust
#[stable(feature = "unsafe_cell_copy_clone_conservative", since = "1.31.0")]
impl<T: Clone + Copy> Clone for UnsafeCell<T> {
    fn clone(&self) -> UnsafeCell<T> {
        UnsafeCell::new(self.value)
    }
}
```

Reply:
Yes that should do it.

Reply:
Together with a comment why we are not calling `clone()`.

Reply:
@RalfJung impl'd in 67bcac6, thanks!

Reply:
Since `Self: Copy` the body of this method can even be just `*self`, though this is equivalent.